### PR TITLE
21027 alembic script and model for batch tables

### PR DESCRIPTION
**Pull request by kzdev420**
*Issue #:* /bcgov/entity#21027

*Description of changes:*

batch.start_date sets up now() default
Screenshot 2024-05-01 at 11 58 28 AM

batch_processing.created_date and last_modified set up now() default
Screenshot 2024-05-01 at 11 59 12 AM

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/models/batch_processing.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    def json(self) -> dict:
+        """Return the batch as a json object."""
+        return {
+            'id': self.id,
+            'step': self.step.name,
+            'status': self.status.name,
+            'createdDate': LegislationDatetime.as_legislation_timezone(self.created_date).isoformat(),
+            'lastUpdated': LegislationDatetime.as_legislation_timezone(self.last_modified).isoformat(),
+            'notes': self.notes
+        }
```
Comment (argush3) [inline legal-api/src/legal_api/models/batch_processing.py:67]: don't need this.  same comment as for batch model

File: legal-api/src/legal_api/models/batch.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        """Return the batch as a json object."""
+        return {
+            'id': self.id,
+            'batch_type': self.batch_type.name,
+            'status': self.status.name,
+            'batchSize': self.size,
+            'startDate': LegislationDatetime.as_legislation_timezone(self.start_date).isoformat(),
+            'endDate': LegislationDatetime.as_legislation_timezone(self.end_date).isoformat(),
+            'notes': self.notes
+        }
```
[the remark this answers is not in the document]
Comment (kzdev420) [reply legal-api/src/legal_api/models/batch.py]: Removed

File: legal-api/src/legal_api/models/batch_processing.py (hunk 1 above)
Comment (kzdev420) [reply legal-api/src/legal_api/models/batch_processing.py]: Removed

File: legal-api/migrations/versions/9fc8fcdc0a22_batches.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+Create Date: 2024-05-01 10:29:37.327296
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = '9fc8fcdc0a22'
+down_revision = '246e135484b3'
```
[the remark this answers is not in the document]
Comment (kzdev420) [reply legal-api/migrations/versions/9fc8fcdc0a22_batches.py]: updated

File: legal-api/src/legal_api/models/batch.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+            query = query.filter(Batch.status == status)
+
+        if size:
+            query = query.filter(Batch.size == size)
+
+        if notes:
+            query = query.filter(Batch.notes == notes)
+
+        batches = query.order_by(Batch.id).all()
+        return batches
```
Comment (argush3) [inline legal-api/src/legal_api/models/batch.py:86]: can we remove `size` and `notes` for now?  Don't see a use case for searching by that yet

File: legal-api/src/legal_api/models/batch_processing.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+            query = query.filter(BatchProcessing.step == step)
+
+        if status:
+            query = query.filter(BatchProcessing.status == status)
+
+        if notes:
+            query = query.filter(BatchProcessing.notes == notes)
+
+        batch_processinges = query.order_by(BatchProcessing.id).all()
+        return batch_processinges
```
Comment (argush3) [inline legal-api/src/legal_api/models/batch_processing.py:96]: can we remove `notes` for now?  Don't see a use case for searching by that yet

File: legal-api/src/legal_api/models/batch.py (hunk 4 above)
Comment (kzdev420) [reply legal-api/src/legal_api/models/batch.py]: Removed

File: legal-api/src/legal_api/models/batch_processing.py (hunk 5 above)
Comment (kzdev420) [reply legal-api/src/legal_api/models/batch_processing.py]: Removed

File: legal-api/src/legal_api/models/batch_processing.py [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+            batch_processing = cls.query.filter_by(id=batch_processing_id).one_or_none()
+        return batch_processing
+
+    @classmethod
+    def find_by(cls,  # pylint: disable=too-many-arguments
+                batch_id: int = None,
+                business_id: int = None,
+                step: BatchProcessingStep = None,
+                status: BatchProcessingStatus = None,
+                notes: str = None) -> dict:
```
Comment (argush3) [inline legal-api/src/legal_api/models/batch_processing.py:75]: can we remove the `notes` parameter too?

File: legal-api/src/legal_api/models/batch.py [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        if batch_id:
+            batch = cls.query.filter_by(id=batch_id).one_or_none()
+        return batch
+
+    @classmethod
+    def find_by(cls,  # pylint: disable=too-many-arguments
+                batch_type: BatchType = None,
+                status: BatchStatus = None,
+                size: int = None,
+                notes: str = None) -> dict:
```
Comment (argush3) [inline legal-api/src/legal_api/models/batch.py:68]: can we remove the `size` and `notes` parameter too?

File: legal-api/src/legal_api/models/batch_processing.py (hunk 6 above)
Comment (kzdev420) [reply legal-api/src/legal_api/models/batch_processing.py]: removed

**Review by argush3 — APPROVED** [review]

**Pull request merged**